GVFS/GVFS.Platform.Mac/MacFileSystem.cs

```diff
@@ -36,12 +36,50 @@ public void ChangeMode(string path, int mode)
         public bool TryGetNormalizedPath(string path, out string normalizedPath, out string errorMessage)
         {
             return MacFileSystem.TryGetNormalizedPathImplementation(path, out normalizedPath, out errorMessage);
+        }
+
+        public bool TryReadFirstByteOfFile(string fileName)
+        {
+            return NativeFileReader.TryReadFirstByteOfFile(fileName);
         }
-
+        
         [DllImport("libc", EntryPoint = "chmod", SetLastError = true)]
         private static extern int Chmod(string pathname, int mode);
 
         [DllImport("libc", EntryPoint = "rename", SetLastError = true)]
         private static extern int Rename(string oldPath, string newPath);
+
+        private class NativeFileReader
+        {
+            public const int ReadOnly = 0x0000;
+
+            public static bool TryReadFirstByteOfFile(string fileName)
+            {
+                int fileDescriptor = Open(fileName, ReadOnly);
+                return TryReadOneByte(fileDescriptor);
+            }
+
+            private static unsafe bool TryReadOneByte(int fileDescriptor)
+            {
+                byte[] buffer = new byte[1];
+                fixed (byte* p = buffer)
+                {
+                    int resultCode = Read(fileDescriptor, p, 1);

```

Comment:
The return is actually `numBytes`, not `resultCode`, right? It just so happens that -1 is a special value.